### PR TITLE
Add csv logging of facts

### PR DESCRIPTION
**Pull request by Williangalvani**
This makes each vehicle save a .csv file named as "<timestamp>vehicle<vehicle_ID>.csv"
This is a sample file (renamed bc github dislikes .csv): [2019-05-21 12-26-41_vehicle1.txt](https://github.com/Williangalvani/qgroundcontrol/files/3203369/2019-05-21.12-26-41_vehicle1.txt)

### CODE REVIEW

**Comment by jaxxzer** [comment]
Is 1kB/s acceptable? (my vote is yes)
A 1hr log would be 3.6MB

File: src/Vehicle/Vehicle.cc [hunk 1]
```diff
@@ -3964,6 +3976,46 @@ void Vehicle::_pidTuningAdjustRates(bool setRatesForTuning)
...
+    if (!_csvLogFile.open(QIODevice::Append)) {
+        qCWarning(VehicleLog) << "unable to open file for csv logging, Stopping csv logging!";
+        _csvLogTimer.stop();
+        return;
+    }
+
+    QTextStream stream(&_csvLogFile);
+
+    // Write CSV header
+    if (firstRun) {
```
Comment (jaxxzer) [inline src/Vehicle/Vehicle.cc]: pull this to a separate function: `Vehicle::_writeCsvHeader`, it is really independent of `writeLine`.

Comment (jaxxzer) [reply src/Vehicle/Vehicle.cc]: On further inspection, I think you should create a function `Vehicle::_initializeCsv` which will create the file and write the header. Call `_initializeCsv` in the vehicle constructor.

File: src/Vehicle/Vehicle.cc [hunk 2]
```diff
@@ -301,6 +304,14 @@ Vehicle::Vehicle(LinkInterface*             link,
     connect(&_adsbTimer, &QTimer::timeout, this, &Vehicle::_adsbTimerTimeout);
     _adsbTimer.setSingleShot(false);
     _adsbTimer.start(1000);
+
+    // Setup csv logging
+    QString now = QDateTime::currentDateTime().toString("yyyy-MM-dd hh-mm-ss");
+    QString fileName = _fileNameTemplate.arg(now).arg(_id);
```
Comment (jaxxzer) [inline src/Vehicle/Vehicle.cc]: You can declare/define the `_fileNameTemplate` here. It is the only place the string is used.

**Comment by jaxxzer** [comment]
looks sick! :clap:

**Comment by jaxxzer** [comment]
Prefix the fact names with the factgroup name. For example, there are multiple `timeRemaining` columns. and the `rotation*` have vague meaning.

**Comment by Williangalvani** [comment]
> Is 1kB/s acceptable? (my vote is yes)
> A 1hr log would be 3.6MB

Seems acceptable to me. That is a tiny percentage of what a video would take. And for mobile, any app update or WhatsApp video is larger than that.

**Comment by Williangalvani** [comment]
@jaxxzer Updated

**Pull request closed without merging**